simplecpp.cpp

```diff
@@ -1303,16 +1330,192 @@ void simplecpp::TokenList::constFoldLogicalOp(Token *tok)
         }
         if (tok->str() != "&&" && tok->str() != "||")
             continue;
-        if (!tok->previous || !tok->previous->number)
+        if (!tok->previous || !tok->next)
             continue;
-        if (!tok->next || !tok->next->number)
+        if (!tok->previous->number && !tok->next->number)
             continue;
 
         int result;
-        if (tok->str() == "||")
-            result = (stringToLL(tok->previous->str()) || stringToLL(tok->next->str()));
-        else /*if (tok->str() == "&&")*/
-            result = (stringToLL(tok->previous->str()) && stringToLL(tok->next->str()));
+        /* std::set<std::string> breakPoints;
```

Comment:
I think your idea here with breakPoints is good. I am thinking that it should be possible to reuse the same code for "||" and "&&" simplifications.